### ISSUE TITLE
Make settings visible to the plugin itself without depending on itself

### ISSUE THREAD

**Issue by frgomes** [tkawachi/sbt-doctest#82]
( Feedback is needed before we spend time implementing anything. )

## Background

At the moment, the plugin in its ``build.sbt`` file adjusts values of variables, like shown below:
```scala
  doctestTestFramework := DoctestTestFramework.ScalaTest,
  doctestMarkdownEnabled := true,
  doctestMarkdownPathFinder := (resourceDirectory in Test).value ** "*.md"
```

At the moment, this is acomplished via this dependency in ``project/plugins.sbt``:
```scala
addSbtPlugin("com.github.tkawachi" % "sbt-doctest"     % "0.4.1")
```
## A bit of guesswork

This may even be the reason behind #81? Something to be double checked later.

**EDIT:** No, it's not the reason behind #81.

## Requirement

As a plugin developer, I wouldn't like to see the plugin including itself as an explicit dependency which could trigger SBT to enter a long chain of dependency resolution.

**EDIT:** Given that the supposition (which stated that it was the reason behind #81) turned out to be wrong, the requirement loses its argument. Yes, makes sense to say that *it looks a bit strange* to have a plugin dependency from itself. Yes, makes sense to say that the same effect could be possibly accomplished in some other way. However, the requirement does not look to be sound anymore.

### Implementation via refactoring

This proposal is about refactoring settings (its *data*) of the plugin into its ``project`` folder, keeping the remaining of the plugin (its *behavior*) as it is. We could simply remove the line in ``project/plugins.sbt`` where it depends on itself. Optionally, the dependency could be moved to a file called ``project/doctest.sbt`` and  ``project/doctest.sbt`` could be added to ``.gitignore``.

### Implementation via configuration

We could wire a call to the plugin itself in the ``build.sbt`` file. Something like this:
```scala
testFrameworks += new TestFramework("doctest.runner.utest.Framework")
```
This way, when a given framework starts, it provides some reasonable defaults necessary for it's own execution. In this example in particular, ``doctest.runner.utest.Framework`` must guarantee that uTest is wired in the ``testFrameworks`` chain, or something equivalent to this:
```scala
testFrameworks += new TestFramework("utest.runner.Framework")
testFrameworks += new TestFramework("doctest.runner.utest.Framework")
```

## Management of dependencies

IMHO, ``sbt-doctest`` should *never* add dependencies into the dependency chain since the developer is in general absolutely oblivious about the inner workings of ``sbt-doctest`` and may get into trouble in case of conflicting dependencies.

In other words, the developer should be *always* responsible for *explicitly* adding the dependencies the developer needs.

**Comment by tkawachi** [comment]
I agree to drop the default dependency to ScalaTest because this plugin tends to introduce an outdated version of  ScalaTest, and developers usually takes care of test framework dependency by themselves.

**Comment by frgomes** [comment]
@tkawachi : I've just discovered that the a SBT plugin always see itself.
Well... this is more or less obvious because the code base sees itself, however SBT honours this fact by inserting the plugin into it's own dependency chain. This is the fact I was oblivious about. Try ``show dependencyList`` or ``show test:dependencyList`` if you are curious.

So, it basically means that the explicit dependency can be removed from ``project/plugins.sbt``. More than that, in the specific case of ``sbt-doctest`` it means that doctests are processed as you would expect by the plugin itself.

**Comment by frgomes** [comment]
@tkawachi : Please see the EDITs I've inserted in the text of the issue. Turns out that this issue lost the vast majority of its reason to exist. However, I'm going to keep it open for the time being so that I can revisit this subject later.

**Comment by frgomes** [comment]
The results of the discussion here are addressed individualy by issues #84 and #85
